Storage/container/matrix.h

```diff
@@ -40,8 +41,7 @@ namespace container {
 struct PairHash {
   template <typename T>
   void hash_combine(std::size_t& seed, const T& v) const {
-    std::hash<T> hasher;
-    seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+    seed ^= CurrentHashFunction<T>()(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
```

Comment:
Note to self: `hash_combine` is probably best to be its own entity in `Bricks/util`, with a simple test too.